### PR TITLE
 support imageframe in python optimizer 

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

support imageframe in python optimizer 

## How was this patch tested?
integration test

### CODE REVIEW

File: pyspark/bigdl/optim/optimizer.py [hunk 1]
```diff
@@ -732,8 +733,12 @@ def set_validation(self, batch_size, val_rdd, trigger, val_method=None):
         """
         if val_method is None:
             val_method = [Top1Accuracy()]
-        callBigDlFunc(self.bigdl_type, "setValidation", self.value, batch_size,
+        if isinstance(val_rdd, RDD):
```
Comment (zhichao-li) [inline pyspark/bigdl/optim/optimizer.py]: if else should only responsible for the method name instead of the entire callBigDLFunc

**Comment by jenniew** [comment]
Please also update model.evaluate to support ImageFrame.

**Comment by jenniew** [comment]
Get null pointer error when training, caused by no mat in ImageFeature. 
Add these lines to filesToImageFrame() in DataSet.scala
val mat = OpenCVMat.fromPixelsBytes(bytes, height, width)
        imf(ImageFeature.mat) = mat
Please check if it is right.

**Comment by jenniew** [comment]
Can you add python support for reading image folder to ImageFrame, that Imageframe transformer can be applied to?

**Comment by SeaOfOcean** [comment]
You should have transformer PixelBytesToMat() after filesToImageFrame(), which will convert bytes to mat

**Comment by SeaOfOcean** [comment]
ImageFrame currently has python API to read a folder of images
see https://bigdl-project.github.io/master/#APIGuide/Data/#imageframe

**Comment by SeaOfOcean** [comment]
@jenniew

**Comment by SeaOfOcean** [comment]
pr validate pass

**Comment by jenniew** [comment]
LGTM

**Comment by SeaOfOcean** [comment]
jenkins pass http://172.168.2.101:8080/view/PR-V(new)/job/BigDL-PR-Validation-v3/219/

**Pull request merged**